### PR TITLE
[v6r14] use better (2D) plot for number of jobs running per pilot

### PR DESCRIPTION
**Pull request by zmathe**
Original:
8bae255422703a8e95ca152e300a4998
New:
20470a33a7abe73ca5cd4fe693dfce4e

### CODE REVIEW

**Comment by chaen** [comment]
Is this really something to go in a patch release ?

**Comment by zmathe** [comment]
A new type of plot has introduced, which is only cerated when we set normalization value. It will not change the current functionality.

File: AccountingSystem/private/DBUtils.py [hunk 1]
```diff
@@ -314,3 +314,16 @@ def _getBucketTotals( self, dataDict ):
           newData[ bt ] = 0.0
         newData[ bt ] += dataDict[ k ][ bt ]
     return newData
+  
+  def _getOverallMaxValue(self, dataDict):
+    """
+    It returns the maximum value in the overall period.
+    :param dict dataDict the format is { key: { time : [value1, value2, ...valueN] } } where key is the parameter what we want to plot, time is the certain period
```
Comment (chaen) [inline AccountingSystem/private/DBUtils.py]: shorter line please

File: AccountingSystem/private/DBUtils.py [hunk 2]
```diff
@@ -314,3 +314,16 @@ def _getBucketTotals( self, dataDict ):
           newData[ bt ] = 0.0
         newData[ bt ] += dataDict[ k ][ bt ]
     return newData
+  
+  def _getOverallMaxValue(self, dataDict):
```
Comment (chaen) [inline AccountingSystem/private/DBUtils.py]: I am not quite sure what you are trying to do with this method. But I *think* that 
```max([ x for y in d.itervalues() for x in y.itervalues() ])``` does what you want no ?

File: Core/Utilities/Graphs/QualityMapGraph.py [hunk 3]
```diff
@@ -66,10 +66,17 @@ def __init__( self, data, ax, prefs, *args, **kw ):
 
     # Setup the colormapper to get the right colors
     self.cmap = LinearSegmentedColormap( 'quality_colormap', cdict, 256 )
-    #self.cmap = cm.RdYlGn
-    self.norms = normalize( 0, 100 )
+    
+    self.cmap = cm.RdYlGn
```
Comment (chaen) [inline Core/Utilities/Graphs/QualityMapGraph.py]: for better reading, I propose to put this into the else clause lower

File: AccountingSystem/private/DBUtils.py (hunk 2 above)
Comment (phicharp) [inline AccountingSystem/private/DBUtils.py]: you can even remove the share brackets ;-)

File: AccountingSystem/private/DBUtils.py [hunk 4]
```diff
@@ -314,3 +314,16 @@ def _getBucketTotals( self, dataDict ):
...
+    """
+    It returns the maximum value in the overall period.
+    :param dict dataDict the format is { key: { time : [value1, value2, ...valueN] } } where key is the parameter what we want to plot, time is the certain period
+    and value1,...valueN is the value.
+    """
+    maxValue = 0.0
+    for i in dataDict.values():
```
Comment (phicharp) [inline AccountingSystem/private/DBUtils.py]: itervalues()

**Comment by atsareg** [comment]
Review OK

**Comment by petricm** [comment]
Just a thought. What about a log scale with scale chosen so that 1 in the middle and the green red coloring?

**Comment by atsareg** [comment]
It is a good starting point, and can be further improved based on the experience. I would prefer just green rather than blue. Tastes differ. I do not think log scale will be better, not so many jobs running per pilot

**Pull request merged**